File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
import sys

input = sys.stdin.readline
A = list(map(str, input().split("-")))

answer = 0


def my_sum(s):
    sum = 0
    temp = s.split("+")
    for i in temp:
        sum += int(i)
    return sum


for i in range(len(A)):
    temp = my_sum(A[i])
    if i == 0:
        answer += temp
    else:
        answer -= temp

print(answer)

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.js
let fs = require('fs');
const input = fs.readFileSync('./dev/stdin').toString().split('\n');

const terms = input[0].split('-');

let sum = 0;

terms.forEach((term, i) => {
  let n = term.includes('+')
    ? term
        .split('+')
        .map(Number)
        .reduce((acc, cur) => (acc += cur), 0)
    : Number(term);
  if (i === 0) sum += n;
  else sum -= n;
});

console.log(sum);
